twilight-lavalink/src/node.rs

```diff
@@ -539,6 +548,78 @@ impl Connection {
         Ok(())
     }
 
+    async fn get_outgoing_endpoint_based_on_event(
+        &self,
+        outgoing: &OutgoingEvent,
+    ) -> Result<(Method, hyper::Uri), NodeError> {
+        let address = self.config.address;
+        tracing::debug!("forwarding event to {}: {outgoing:?}", address,);
+
+        let guild_id = OutgoingEvent::guild_id(outgoing);
+        let no_replace = OutgoingEvent::no_replace(outgoing);
+
+        if let Some(session) = self.lavalink_session_id.lock().await.clone() {
+            tracing::debug!(
+                "Found session id {}. Generating the url and method for event type.",
+                session
+            );
+
+            if let OutgoingEvent::Destroy(_) = outgoing.clone() {
+                let destroy_uri = Uri::builder()
+                    .scheme("http")
+                    .authority(address.to_string())
+                    .path_and_query(format!("/v4/sessions/{session}/players/{guild_id}"))
+                    .build()
+                    .unwrap();
+                return Ok((Method::DELETE, destroy_uri));
+            }
+            let uri = Uri::builder()
+                .scheme("http")
+                .authority(address.to_string())
+                .path_and_query(format!(
+                    "/v4/sessions/{session}/players/{guild_id}?noReplace={no_replace}"
+                ))
+                .build()
+                .unwrap();
+            return Ok((Method::PATCH, uri));
+        }
+
+        tracing::error!("No session id is found. Session id should have been provided from the websocket connection already.");
+        Err(NodeError {
+            kind: NodeErrorType::OutgoingEventHasNoSession,
+            source: None,
+        })
+    }
+
+    async fn outgoing(&self, outgoing: OutgoingEvent) -> Result<(), NodeError> {
+        let (method, url) = self.get_outgoing_endpoint_based_on_event(&outgoing).await?;
+        let payload = serde_json::to_string(&outgoing).unwrap();
+
+        tracing::debug!("Sending request to {url:?} using method {method:?}.");
+        tracing::debug!("converted payload: {payload:?}");
+
+        let authority = url.authority().expect("Authority comes from endpoint. We should have a valid authority and is just used in the header.").clone();
+
+        let req = Request::builder()
+            .uri(url)
+            .method(method)
+            .header(hyper::header::HOST, authority.as_str())
+            .header(AUTHORIZATION, self.config.authorization.as_str())
+            .header("Content-Type", "application/json")
+            .body(Full::from(payload))
+            .map_err(|source| NodeError {
+                kind: NodeErrorType::BuildingConnectionRequest,
+                source: Some(Box::new(source)),
+            })?;
+
+        tracing::debug!("Request: {req:?}");
+
+        let response = self.lavalink_http.request(req).await.unwrap();
```

Comment:
This may panic

Reply:
Agreed. Changed to a new error. Please let me know what you think.